From db9cb0b9b79647fd880e90e940d7c60e9bb790e2 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Thu, 13 Dec 2012 23:41:12 +0100
Subject: CLEANUP: poll: remove a useless double-check on fdtab[fd].owner

This check is already performed a few lines above in the same loop,
remove it from the condition.
---
 src/ev_epoll.c  |    2 +-
 src/ev_poll.c   |    2 +-
 src/ev_select.c |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ev_epoll.c b/src/ev_epoll.c
index 8b0d684..a9b8410 100644
--- a/src/ev_epoll.c
+++ b/src/ev_epoll.c
@@ -161,7 +161,7 @@ REGPRM2 static void _do_poll(struct poller *p, int exp)
 				((e & EPOLLHUP) ? FD_POLL_HUP : 0);
 		}
 
-		if (fdtab[fd].iocb && fdtab[fd].owner && fdtab[fd].ev) {
+		if (fdtab[fd].iocb && fdtab[fd].ev) {
 			int new_updt, old_updt = fd_nbupdt; /* Save number of updates to detect creation of new FDs. */
 
 			/* Mark the events as speculative before processing
diff --git a/src/ev_poll.c b/src/ev_poll.c
index 2c7501d..c09232e 100644
--- a/src/ev_poll.c
+++ b/src/ev_poll.c
@@ -170,7 +170,7 @@ REGPRM2 static void _do_poll(struct poller *p, int exp)
 				((e & POLLHUP) ? FD_POLL_HUP : 0);
 		}
 
-		if (fdtab[fd].iocb && fdtab[fd].owner && fdtab[fd].ev) {
+		if (fdtab[fd].iocb && fdtab[fd].ev) {
 			/* Mark the events as speculative before processing
 			 * them so that if nothing can be done we don't need
 			 * to poll again.
diff --git a/src/ev_select.c b/src/ev_select.c
index 356eed7..48a62da 100644
--- a/src/ev_select.c
+++ b/src/ev_select.c
@@ -156,7 +156,7 @@ REGPRM2 static void _do_poll(struct poller *p, int exp)
 			if (FD_ISSET(fd, tmp_evts[DIR_WR]))
 				fdtab[fd].ev |= FD_POLL_OUT;
 
-			if (fdtab[fd].iocb && fdtab[fd].owner && fdtab[fd].ev) {
+			if (fdtab[fd].iocb && fdtab[fd].ev) {
 				/* Mark the events as speculative before processing
 				 * them so that if nothing can be done we don't need
 				 * to poll again.
-- 
1.7.1

